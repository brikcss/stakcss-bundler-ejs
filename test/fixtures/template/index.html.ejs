<%
	data.page = Object.assign({
		title: 'EJS compile example',
		description: 'This example illustrates using the vanilla JS flavor of spinner as an <em>ES module, compiled with webpack</em>.',
		content: './_partial.ejs'
	}, data.page || {});
	var stylesheets = stylesheets || [];
	var scripts = scripts || [];
_%>
<!DOCTYPE html>
<html>
<head>
	<title><%= data.page.title %></title>
	<%_ const stylesHtml = []; data.stylesheets.forEach(stylesheet => {
		stylesHtml.push(`<link rel="stylesheet" type="text/css" href="${stylesheet}">`);
	}); _%>
	<%- stylesHtml.join('\n\t'); %>
</head>
<body>
	<div class="page">
		<div class="page__intro">
			<h1 class="page__title"><%= data.page.title %></h1>
			<p style="margin-bottom: 3rem;"><%- data.page.description %></p>
		</div>

		<%- include(data.page.content); _%>
	</div><% _%>

	<%_ const scriptsHtml = []; data.scripts.forEach(scriptPath => {
		scriptsHtml.push(`<script type="text/javascript" src="${scriptPath}"></script>`);
	}); %>
	<%- scriptsHtml.join('\n\t'); %>
</body>
</html>
