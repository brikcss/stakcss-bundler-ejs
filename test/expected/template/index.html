<!DOCTYPE html>
<html>
<head>
	<title>EJS compile example</title>
	<link rel="stylesheet" type="text/css" href="vendor-styles.css">
	<link rel="stylesheet" type="text/css" href="my-library-styles.css">
	<link rel="stylesheet" type="text/css" href="my-styles.css">
</head>
<body>
	<div class="page">
		<div class="page__intro">
			<h1 class="page__title">EJS compile example</h1>
			<p style="margin-bottom: 3rem;">This is <em>my page</em> Yay!!!.</p>
		</div>

		<p>I am the HTML content.</p>
	</div>

	<script type="text/javascript" src="vendor-scripts.js"></script>
	<script type="text/javascript" src="my-library-scripts.js"></script>
	<script type="text/javascript" src="my-view-scripts.js"></script>
</body>
</html>
